File: auth/src/main/java/com/firebase/ui/auth/util/data/SessionUtils.java
/*
 * Copyright 2025 Google Inc. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the
 * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.firebase.ui.auth.util.data;

import java.util.Random;

import androidx.annotation.RestrictTo;

@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
public class SessionUtils {

    private static final String VALID_CHARS =
            "1234567890abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";

    /**
     * Generates a random alpha numeric string.
     * @param length the desired length of the generated string.
     * @return a randomly generated string with the desired number of characters.
     */
    public static String generateRandomAlphaNumericString(int length) {
        StringBuilder sb = new StringBuilder();
        Random random = new Random();
        for (int i = 0; i < length; i++) {
            sb.append(VALID_CHARS.charAt(random.nextInt(length)));
        }
        return sb.toString();
    }
}

File: auth/src/main/java/com/firebase/ui/auth/util/SessionUtils.kt
package com.firebase.ui.auth.util

import androidx.annotation.RestrictTo
import kotlin.random.Random

/**
 * Utility for generating random session identifiers.
 */
@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
object SessionUtils {

    private const val VALID_CHARS = "1234567890abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"

    /**
     * Generates a random alphanumeric string.
     *
     * @param length The desired length of the generated string.
     * @return A randomly generated string with the desired number of characters.
     */
    fun generateRandomAlphaNumericString(length: Int): String {
        return (1..length)
            .map { VALID_CHARS[Random.nextInt(VALID_CHARS.length)] }
            .joinToString("")
    }
}